pageserver/src/layered_repository/timeline.rs

```diff
@@ -391,6 +391,29 @@ pub struct Timeline {
     repartition_threshold: u64,
 
     /// Current logical size of the "datadir", at the last LSN.
+    current_logical_size: LogicalSize,
+
+    /// Information about the last processed message by the WAL receiver,
+    /// or None if WAL receiver has not received anything for this timeline
+    /// yet.
+    pub last_received_wal: Mutex<Option<WalReceiverInfo>>,
+
+    /// Relation size cache
+    pub rel_size_cache: RwLock<HashMap<RelTag, (Lsn, BlockNumber)>>,
+}
+
+/// Internal structure to hold all data needed for logical size calculation.
+/// Calculation consists of two parts: initial size calculation, that potentially might take a lot of time,
+/// since requires all layers up to the `initial_part_end` to be read.
```

Comment:
Not all layers, only the ones containing relation sizes. The point is still valid that it could take a long time, though